tf_quant_finance/black_scholes/vanilla_prices_test.py

```diff
@@ -334,6 +334,144 @@ def test_binary_vanilla_consistency_exact(self):
         tf.where(is_call_options, -implied_binary_price, implied_binary_price))
     self.assertArrayNear(implied_binary_price, actual_binary_price, 1e-10)
 
+  def test_barrier_option_1d_puo(self):
```

Comment:
Could you please test for double precision as well? Please use parametrized test for that. See, e.g., [here](https://github.com/google/tf-quant-finance/blob/1239ce3a22b4a6545d1600835333e8c9984ade7d/tf_quant_finance/black_scholes/brownian_bridge_test.py#L28)

Reply:
Done!